op-node/rollup/derive/pipeline.go

```diff
@@ -96,18 +96,18 @@ func NewDerivationPipeline(log log.Logger, cfg *rollup.Config, l1Fetcher L1Fetch
 
 	// Pull stages
 	l1Traversal := NewL1Traversal(log, l1Fetcher)
+	dataSrc := NewCalldataSource(log, cfg, l1Fetcher) // auxiliary stage for L1Retrieval
```

Comment:
More like a data structure / helper? Should be hidden in the implem of NewL1Retrieval — it's never used anywhere else!

Maybe for testing? Mocking the l1Fetch should be enough.